astropy/coordinates/transformations.py

```diff
@@ -70,6 +70,20 @@ def frame_set(self):
 
         return self._cached_frame_set.copy()
 
+    @property
+    def frame_attrnames_set(self):
+        """
+        A `frozenset` of all the attribute names of all frame classes in this `TransformGraph`.
+        """
+        if self._cached_frame_attrnames_set is None:
+            result = set()
+            for frame_cls in self.frame_set:
+                for attr in frame_cls.get_frame_attr_names().keys():
```

Comment:
That should work (more precisely `result.update(frame_cls.get_frame_attr_names().keys())` ).  but I was relocating known-good code rather than rewriting.

Since this is typically called only once over the life of a program, the performance hit is negligible, although .update does improve readability.